inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongClientImpl.java

```diff
@@ -97,16 +101,41 @@ public InlongGroup forGroup(InlongGroupConf groupConf) {
         }
     }
 
+    @Override
+    public Response<PageInfo<InlongGroupListResponse>> listGroup(InlongGroupPageRequest request) throws Exception {
+        InnerInlongManagerClient managerClient = new InnerInlongManagerClient(this);
+        return managerClient.listGroups(request);
+    }
+
     /**
-     * List group
+     * List group state
      *
-     * @param request The request
-     * @return PageInfo of group
+     * @param groupNames
+     * @return
+     * @throws Exception
      */
     @Override
-    public Response<PageInfo<InlongGroupListResponse>> listGroup(InlongGroupPageRequest request) throws Exception {
+    public Map<String, InlongGroupState> listGroupState(List<String> groupNames) throws Exception {
         InnerInlongManagerClient managerClient = new InnerInlongManagerClient(this);
-        return managerClient.listGroups(request);
+        InlongGroupPageRequest request = new InlongGroupPageRequest();
+        request.setNameList(groupNames);
+        request.setPageNum(1);
+        request.setPageSize(groupNames.size());
+        request.setListSources(true);
+        Response<PageInfo<InlongGroupListResponse>> pageInfoResponse = managerClient.listGroups(request);
+        if (!pageInfoResponse.isSuccess() || pageInfoResponse.getErrMsg() != null) {
+            throw new RuntimeException("listGroupStateFailed:" + pageInfoResponse.getErrMsg());
+        }
+        List<InlongGroupListResponse> groupListResponses = pageInfoResponse.getData().getList();
+        Map<String, InlongGroupState> groupStateMap = Maps.newHashMap();
+        groupListResponses.stream().forEach(groupListResponse -> {
+            String groupId = groupListResponse.getInlongGroupId();
+            InlongGroupState groupState = InlongGroupState.parseByBizStatus(groupListResponse.getStatus());
+            List<SourceListResponse> sourceListResponses = groupListResponse.getSourceListResponses();
```

Comment:
Here we need to check if the List is empty.

Reply:
ok, but groupList is select in mybatis, if no result is return, mybatis return empty list rather than null